src/NzbDrone.Core/Applications/Sonarr/SonarrV3Proxy.cs

```diff
@@ -102,7 +102,17 @@ public ValidationFailure TestConnection(SonarrIndexer indexer, SonarrSettings se
 
             try
             {
-                Execute<SonarrIndexer>(request);
+                var version = _httpClient.Post<SonarrIndexer>(request).Headers.GetSingleValue("X-Application-Version");
+
+                if (version == null)
+                {
+                    return new ValidationFailure(string.Empty, "Failed to fetch Sonarr version");
+                }
+
+                if (new Version(version) < new Version("3.0.9.0"))
```

Comment:
```suggestion
                if (new Version(version) < new Version("3.0.9.1549"))
```

Reply:
let's move the version to a var to make it easier to maintain

Reply:
builds have never - in the ~12 years of starr have never been reset within a major version